openmoc/map_to_dict.i

```diff
@@ -126,13 +126,65 @@
     }
 
     /* Loop over y */
-    for (int j = 0; j < $2; j++) {
+    for (int i =0; i < $2; i++) {
       /* Extract the value from the list at this location and convert
        * SWIG wrapper to pointer to underlying C++ class instance */
-      PyObject* o = PyList_GetItem(outer_list, j);
+      PyObject* o = PyList_GetItem(outer_list, i);
       void *p1 = 0;
       SWIG_ConvertPtr(o, &p1, SWIGTYPE_p_Universe, 0 | 0);
-      $3[i*$2+j] = (Universe*) p1;
+      $3[j*$2+i] = (Universe*) p1;
+    }
+  }
+}
+
+
+/* Typemap for Lattice::setUniverses3D(int num_z, int num_y, int num_x, 
```

Comment:
I'm fine with that as well.
On Oct 12, 2015 6:43 PM, "sam" <notifications@github.com> wrote:

> In openmoc/map_to_dict.i
> <https://github.com/mit-crpg/OpenMOC/pull/188#discussion_r41809295>:
>
> >        void *p1 = 0;
> >        SWIG_ConvertPtr(o, &p1, SWIGTYPE_p_Universe, 0 | 0);
> > -      $3[i*$2+j] = (Universe*) p1;
> > +      $3[j*$2+i] = (Universe*) p1;
> > +    }
> > +  }
> > +}
> > +
> > +
> > +/* Typemap for Lattice::setUniverses3D(int num_z, int num_y, int num_x,
>
> I'm on board with this, but if this is done I'm in favor of changing the
> Circle class to ZCylinder to make it clear that we are requiring
> geometries to be 3D instead of assuming they are axially extruded.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/mit-crpg/OpenMOC/pull/188/files#r41809295>.
>